src/packages/EmbedIO/WebApi/FormFieldAttribute.cs

```diff
@@ -161,9 +162,21 @@ private FormFieldAttribute(bool badRequestIfMissing, string? fieldName)
             {
                 var fieldValue = data.GetValues(fieldName)?.LastOrDefault();
                 if (fieldValue == null)
-                    return type.IsValueType ? Activator.CreateInstance(type) : null;
-
-                if (!FromString.TryConvertTo(type, fieldValue, out var result))
+                {
+                    if (type.IsValueType)
+                    {
+                        result = Activator.CreateInstance(type);
+                        var parameter = controller.CurrentMethod.GetParameters().FirstOrDefault(p => p.Name == fieldName);
+                        if (parameter.HasDefaultValue)
+                        {
+                            result = parameter.DefaultValue;
+                        }
```

Comment:
`Activator.CreateInstance` is slow, call it only if necessary:

* remove the early initialization of `result` in line 168
* replace this
```C#
                        if (parameter.HasDefaultValue)
                        {
                            result = parameter.DefaultValue;
                        }
```
with this:
```C#
                        result = parameter.HasDefaultValue
                            ? parameter.DefaultValue
                            : Activator.CreateInstance(type);
```